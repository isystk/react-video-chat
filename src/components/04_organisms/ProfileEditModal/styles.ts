import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles((theme) => ({
  notion: {
    width: '300px',
    '& .myHeadPhoto': {
      position: 'relative',
      width: '100px',
      height: '100px',
      // borderRadius: '50%',
      margin: '0 auto',
      overflow: 'hidden',
      '& .photoEditBtn': {
        position: 'absolute',
        right: '0',
        bottom: '0',
        fontSize: '15px',
        background: '#fff',
        opacity: '0.7',
        textDecoration: 'none'
      }
    },
    '& .myHeadPhoto img': {
      width: '100%',
    },
    '& .myName': {
      width: '100%',
      fontSize: '24px',
      color: '#222222',
      textAlign: 'center',
      marginTop: '15px',
    },
    '& .btn': {
      width: '100%',
      textAlign: 'center',
    },
    '& .btn button': {
      margin: '25px 25px 5px',
    },
  },
}))

export { useStyles }
