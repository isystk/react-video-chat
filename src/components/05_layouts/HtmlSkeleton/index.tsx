import React, { Children, FC, useState } from 'react'
import { connect } from '@/components/hoc'
import { ContainerProps, WithChildren } from 'types'
import useAppRoot from '@/stores/useAppRoot'
import Header from '@/components/04_organisms/Header'
import SideMenu from '@/components/04_organisms/SideMenu'
import Title, { TitleProps } from './Title'
import Head from 'next/head'
import { isReactElement } from '@/utils/general/object'
import { CssBaseline } from '@material-ui/core'

/** HtmlSkeleton Props */
export type HtmlSkeletonProps = WithChildren
/** Presenter Props */
export type PresenterProps = HtmlSkeletonProps & {
  title?: TitleProps['children']
  description: string
}

/** Presenter Component */
const HtmlSkeletonPresenter: FC<PresenterProps> = ({
  main,
  title,
  description,
  isMenuOpen,
  setMenuOpen,
  children,
  ...props
}) => (
  <>
    <Head>
      {/* タイトル */}
      <title>{title}</title>
      {/* favicon */}
      <link rel="icon" href="/favicon.ico" />
      {/* ブラウザテーマカラー */}
      <meta name="theme-color" content="#ffffff" />
      {/* サイト概要 */}
      <meta name="description" content={description} />
      {/* OGP 画像URL */}
      <meta property="og:image" content={'/ogp-image.png'} />
      {/* OGP タイトル */}
      <meta name="og:title" content={title} />
      {/* OGP サイト概要 */}
      <meta name="og:description" content={description} />
      {/* OGP Twitterカード */}
      <meta name="twitter:card" content="summary" />
      {/* apple ポータブル端末 アイコン */}
      <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
      {/* manifest.json */}
      <link rel="manifest" href="/manifest.json" />
    </Head>
    <Context.Provider value={main}>
      <CssBaseline />
      <Header isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
      <div>{children}</div>
      <SideMenu isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
    </Context.Provider>
  </>
)

/** Container Component */
const HtmlSkeletonContainer: React.FC<
  ContainerProps<HtmlSkeletonProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  let title: TitleProps['children'] | undefined = undefined
  const main = useAppRoot()
  const [isMenuOpen, setMenuOpen] = useState(false)

  if (!main) return <></>

  const APP_NAME = process.env.APP_NAME

  children = Children.map(children, (child) =>
    isReactElement(child) && child.type === Title
      ? (title = `${child.props.children} | ${APP_NAME}`) && undefined
      : React.cloneElement(child)
  )
  if (!title) title = process.env.APP_NAME

  const description = `${APP_NAME} is an application that allows you to casually chat with your friends!`
  return presenter({
    main,
    title,
    description,
    isMenuOpen,
    setMenuOpen,
    children,
    ...props,
  })
}

export const Context = React.createContext(null)

export default connect<HtmlSkeletonProps, PresenterProps>(
  'HtmlSkeleton',
  HtmlSkeletonPresenter,
  HtmlSkeletonContainer
)

// Sub Component
export type { TitleProps }
export { Title }
