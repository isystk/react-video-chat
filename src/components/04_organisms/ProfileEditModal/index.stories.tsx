import { Meta, Story } from '@storybook/react'
import React from 'react'
import ProfileEditModal from './index'
import MainService from '@/services/main'
import { Context } from '@/components/05_layouts/HtmlSkeleton'

export default {
  title: '04_organisms/ProfileEditModal',
  component: ProfileEditModal,
} as Meta

const Template: Story = (props) => {
  const main = new MainService(() => ({}))
  main.setName('isystk')
  main.room.rooms = { '1': { id: '1', name: 'test', description: 'test' } }
  main.room.setRoomId('1')
  main.self.isOpen = true
  return (
    <Context.Provider value={main}>
      <ProfileEditModal />
    </Context.Provider>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
