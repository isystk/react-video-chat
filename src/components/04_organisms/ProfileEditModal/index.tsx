import React, { FC, useContext, useState } from 'react'
import Modal from '@/components/01_atoms/Modal'
import { Button } from '@material-ui/core'
import Container from '@material-ui/core/Container'
import { ContainerProps, WithChildren } from 'types'
import { useStyles } from './styles'
import { connect } from '@/components/hoc'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import MainService from '@/services/main'
import { imageCompression, fileToDataURL } from '@/utils/general'

/** ProfileEditModalProps Props */
export type ProfileEditModalProps = WithChildren
/** Presenter Props */
export type PresenterProps = ProfileEditModalProps & {
  main
  classes
  isOpen
  connectionId
  name
  photo
}

/** Presenter Component */
const ProfileEditModalPresenter: FC<PresenterProps> = ({
  main,
  classes,
  isOpen,
  connectionId,
  name,
  photo,
  handleClose,
  editProfilePhoto,
  ...props
}) => (
  <>
    <Modal isOpen={isOpen} hideCloseBtn={true}>
      <Container component="main">
        <div className={classes.notion}>
          <div className="myHeadPhoto">
            <img src={photo} alt="" />
            <label className="photoEditBtn" onChange={editProfilePhoto}>
              <input type="file" />
              編集
            </label>
          </div>
          <div className="myName">{name}</div>
          <div className="btn">
            <Button
              color="secondary"
              onClick={handleClose}
              type="submit"
              variant="contained"
            >
              キャンセル
            </Button>
            <Button
              color="primary"
              onClick={(e) => main.storeProfile({ photo })}
              type="submit"
              variant="contained"
            >
              保存する
            </Button>
          </div>
        </div>
      </Container>
    </Modal>
  </>
)

/** Container Component */
const ProfileEditModalContainer: React.FC<
  ContainerProps<ProfileEditModalProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const main = useContext<MainService | null>(Context)
  if (!main) return <></>
  const classes = useStyles()

  const [photo, setPhoto] = useState<string>(main.self.photo)

  const editProfilePhoto = async (e) => {
    const file = e.target.files[0]
    // 画像を圧縮する
    const cfile = await imageCompression(file)
    // 画像をBase64に変換する
    const base64 = await fileToDataURL(cfile)
    setPhoto(base64)
  }

  const handleClose = () => {
    setPhoto(main.self.photo)
    main.closeProfileEdit()
  }

  const { connectionId, name, isOpen } = main.self
  return presenter({
    children,
    main,
    classes,
    isOpen,
    connectionId,
    name,
    photo,
    handleClose,
    editProfilePhoto,
    ...props,
  })
}

export default connect<ProfileEditModalProps, PresenterProps>(
  'ProfileEditModal',
  ProfileEditModalPresenter,
  ProfileEditModalContainer
)
