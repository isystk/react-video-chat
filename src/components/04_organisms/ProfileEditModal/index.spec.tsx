import React from 'react'
import renderer from 'react-test-renderer'
import ProfileEditModal from './index'
import '@testing-library/jest-dom/extend-expect'
import MainService from '@/services/main'
import { Context } from '@/components/05_layouts/HtmlSkeleton'

describe('ProfileEditModal', () => {
  it('Match Snapshot', () => {
    const main = new MainService(() => ({}))
    main.setName('isystk')
    main.setRoomId('test')
    main.self.isOpen = true
    const component = renderer.create(
      <Context.Provider value={main}>
        <ProfileEditModal />
      </Context.Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
