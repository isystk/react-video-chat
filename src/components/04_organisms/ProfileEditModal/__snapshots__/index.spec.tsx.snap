// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileEditModal Match Snapshot 1`] = `null`;
